src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonService.kt

```diff
@@ -10,27 +10,25 @@ import javax.persistence.EntityNotFoundException
 const val CLIENT_CAN_MAINTAIN_EMAIL_ADDRESSES = "hasRole('MAINTAIN_REF_DATA') and hasAuthority('SCOPE_write')"
 
 @Service
+@Transactional(readOnly = true)
 class PrisonService(
   private val prisonRepository: PrisonRepository,
   private val videoLinkConferencingCentreRepository: VideoLinkConferencingCentreRepository,
   private val offenderManagementUnitRepository: OffenderManagementUnitRepository
 ) {
-  @Transactional(readOnly = true)
-  fun findById(prisonId: String): PrisonDto {
-    val prison =
-      prisonRepository.findByIdOrNull(prisonId) ?: throw EntityNotFoundException("Prison $prisonId not found")
-    return PrisonDto(prison)
-  }
+  fun findById(prisonId: String): Prison =
+    prisonRepository.findByIdOrNull(prisonId) ?: throw EntityNotFoundException("Prison $prisonId not found")
+
+  fun findByGpPractice(gpPracticeCode: String): Prison =
+    prisonRepository.findByGpPracticeGpPracticeCode(gpPracticeCode)
+      ?: throw EntityNotFoundException("Prison with gp practice $gpPracticeCode not found")
```

Comment:
I couldn't find a test for the `throw` here - might have missed it though :smile:

Reply:
will fix in next PR - I copied verbatim from prison estate so entirely likely missed a test along the way